scripts/archive_docs.py

```diff
@@ -79,10 +80,32 @@ def list_tree(source):
 		output = [os.path.basename(x) for x in output if len(x) > 0]
 		return output
 
-def copy_file(source_path, target_path):
+def copy_file(source_path, target_path, version):
 	print(f"{source_path} -> {target_path}")
 	if revision == None:
-		shutil.copy(source_path, target_path)
+		if source_path.endswith(".md"):
+			with open(source_path) as f, open(target_path, "w") as of:
+				# parse YAML metadata and adjust "redirect_from" fields
+				doc_text = f.read()
+				doc_groups = doc_text.split("---\n")
```

Comment:
Rather than copying the frontmatter parsing and generation code, perhaps we should switch to https://pypi.org/project/frontmatter/ or move the code into a shared package?